File: src/main/java/io/rsocket/cli/SystemInCharSource.java
package io.rsocket.cli;

import com.google.common.io.CharSource;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.Reader;

public class SystemInCharSource extends CharSource {
  public static final CharSource INSTANCE = new SystemInCharSource();

  private SystemInCharSource() {}

  @Override
  public Reader openStream() throws IOException {
    return new InputStreamReader(System.in);
  }
}

File: src/main/kotlin/io/rsocket/cli/SystemInCharSource.kt
package io.rsocket.cli

import com.google.common.io.CharSource
import java.io.IOException
import java.io.InputStreamReader
import java.io.Reader

class SystemInCharSource private constructor() : CharSource() {

    @Throws(IOException::class)
    override fun openStream(): Reader = InputStreamReader(System.`in`)

    companion object {
        val INSTANCE: CharSource = SystemInCharSource()
    }
}
